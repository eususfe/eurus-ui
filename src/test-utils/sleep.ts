/*
 * @Author: jiangxd
 * @Date: 2022-09-23 14:06:55
 * @LastEditTime: 2022-09-23 14:07:04
 * @LastEditors: jiangxd
 * @Description:
 * @FilePath: /eurus-ui/src/test-utils/sleep.ts
 */
const sleep = (time = 0) => new Promise(resolve => setTimeout(resolve, time));

export default sleep;
