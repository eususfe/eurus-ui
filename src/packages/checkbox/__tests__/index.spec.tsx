import { mount } from '@vue/test-utils';
import { h } from 'vue';
import { expect } from 'vitest';
import ECheckbox from '..';
import ECheckboxGroup from '../src/checkbox-group';

describe('ECheckbox', () => {

  it('ECheckbox snapshot', () => {
    const wrapper = mount(ECheckbox);
    expect(wrapper.html()).toMatchSnapshot();
    wrapper.unmount();
  });

  it('should emit change event', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckbox, {
      props: {
        modelValue: false,
        onChange: fn

      }
    }
    );
    await wrapper.find('input').trigger('change');
    expect(fn).toHaveBeenCalled();
  });

  it('shod not emit change event when disabled', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckbox, {
      props: {
        modelValue: false,
        onChange: fn,
        disabled: true
      }
    });
    await wrapper.find('input').trigger('change');
    expect(fn).not.toHaveBeenCalled();
  });

  it('should work emit change event in group', async () => {
    const fn = vi.fn();
    const wrapper = mount(ECheckboxGroup, {
      props: {
        modelValue: ['1'],
        onChange: fn
      },
      slots: {

        default: () => h(ECheckbox, { label: '1', value: '1' })
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
    await wrapper.findComponent(ECheckbox).find('input').trigger('change');
    expect(fn).toHaveBeenCalled();
  });

  it('should work with label props', () => {
    const wrapper = mount(ECheckbox, {
      props: {
        label: 'test'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.eu-checkbox__label').text()).toBe('test');
  });
});

