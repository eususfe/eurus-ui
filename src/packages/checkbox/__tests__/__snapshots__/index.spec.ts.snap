// Vitest Snapshot v1

exports[`ECheckbox > ECheckbox snapshot 1`] = `
"<div class=\\"eu-checkbox\\" aria-hidden=\\"true\\"><label class=\\"\\"><input class=\\"\\" type=\\"checkbox\\">
    <div class=\\"slot-text\\">
      <!---->
    </div>
  </label></div>"
`;
